packages/material-ui/src/Select/Select.js

```diff
@@ -81,11 +83,12 @@ const Select = React.forwardRef(function Select(inProps, ref) {
             SelectDisplayProps: { id, ...SelectDisplayProps },
           }),
       ...inputProps,
-      classes: inputProps ? deepmerge(classes, inputProps.classes) : classes,
+      classes: inputProps ? deepmerge(otherClasses, inputProps.classes) : otherClasses,
       ...(input ? input.props.inputProps : {}),
     },
     ...(multiple && native && variant === 'outlined' ? { notched: true } : {}),
     ref,
+    classes: { root: rootClassName },
```

Comment:
It makes me think of https://github.com/mui-org/material-ui/issues/25313#issuecomment-834816185. I'm not sure we need to solve this now, but if we do, we might as well do it completely (update the tests & cover NativeSelect).

Reply:
I updated this PR to do it correctly. It's part of the polishing.